forem/forem
# Fix Facebook Login profile image download failure

## What type of PR is this? (check all applicable)

## Description
This fixes a bug with Facebook Login where you can't create an account because the profile image download process fails. The workaround is:
1. Grab the user's Facebook UID and OAuth access token
2. Request the photo directly via a separate call
3. From that JSON response, grab the direct link to the requested profile image
4. Pass along to Carrierwave to download as normal

## Related Tickets & Documents
Closes https://github.com/forem/InternalProjectPlanning/issues/238

## QA Instructions, Screenshots, Recordings
1. Make a Facebook OAuth app https://docs.forem.com/backend/auth-facebook/
2. Add your Facebook keys to your config
3. Try to login via Facebook
4. See that your user creates successfully

### UI accessibility concerns?
Nope!

## Added tests?
- [x] No, and this is why: it's hard to stub this, and I think this is more of an implementation detail

## Added to documentation?

## [optional] Are there any post deployment tasks we need to perform?
Nope

## [optional] What gif best describes this PR or how it makes you feel?

![Alice falls through the rabbit hole.](https://media.giphy.com/media/swtiK9jRfE0zS/giphy.gif)
